File: app/build_backup/tmp/kapt3/stubs/debug/com/chilluminati/rackedup/data/database/dao/UserProfileDao.java
package com.chilluminati.rackedup.data.database.dao;

@kotlin.Metadata(mv = {1, 9, 0}, k = 1, xi = 48, d1 = {"\u0000 \n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\t\n\u0002\b\u0003\n\u0002\u0010\u0002\n\u0000\bg\u0018\u00002\u00020\u0001J\u0010\u0010\u0002\u001a\u0004\u0018\u00010\u0003H\u00a7@\u00a2\u0006\u0002\u0010\u0004J\u0016\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\u0003H\u00a7@\u00a2\u0006\u0002\u0010\bJ\u0016\u0010\t\u001a\u00020\n2\u0006\u0010\u0007\u001a\u00020\u0003H\u00a7@\u00a2\u0006\u0002\u0010\b\u00a8\u0006\u000b"}, d2 = {"Lcom/chilluminati/rackedup/data/database/dao/UserProfileDao;", "", "getActiveProfile", "Lcom/chilluminati/rackedup/data/database/entity/UserProfile;", "(Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "insertProfile", "", "profile", "(Lcom/chilluminati/rackedup/data/database/entity/UserProfile;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;", "updateProfile", "", "app_debug"})
@androidx.room.Dao()
public abstract interface UserProfileDao {
    
    @androidx.room.Query(value = "SELECT * FROM user_profiles WHERE is_active = 1 LIMIT 1")
    @org.jetbrains.annotations.Nullable()
    public abstract java.lang.Object getActiveProfile(@org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super com.chilluminati.rackedup.data.database.entity.UserProfile> $completion);
    
    @androidx.room.Insert(onConflict = 1)
    @org.jetbrains.annotations.Nullable()
    public abstract java.lang.Object insertProfile(@org.jetbrains.annotations.NotNull()
    com.chilluminati.rackedup.data.database.entity.UserProfile profile, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super java.lang.Long> $completion);
    
    @androidx.room.Update()
    @org.jetbrains.annotations.Nullable()
    public abstract java.lang.Object updateProfile(@org.jetbrains.annotations.NotNull()
    com.chilluminati.rackedup.data.database.entity.UserProfile profile, @org.jetbrains.annotations.NotNull()
    kotlin.coroutines.Continuation<? super kotlin.Unit> $completion);
}

File: app/src/main/java/com/chilluminati/rackedup/data/database/dao/UserProfileDao.kt
package com.chilluminati.rackedup.data.database.dao

import androidx.room.*
import com.chilluminati.rackedup.data.database.entity.UserProfile
import kotlinx.coroutines.flow.Flow

@Dao
interface UserProfileDao {
    @Query("SELECT * FROM user_profiles WHERE is_active = 1 LIMIT 1")
    suspend fun getActiveProfile(): UserProfile?
    
    @Query("SELECT * FROM user_profiles WHERE is_active = 1 LIMIT 1")
    fun getActiveProfileFlow(): Flow<UserProfile?>
    
    @Query("SELECT * FROM user_profiles")
    suspend fun getAllProfiles(): List<UserProfile>
    
    @Query("SELECT * FROM user_profiles ORDER BY created_at ASC")
    fun getAllProfilesFlow(): Flow<List<UserProfile>>
    
    @Query("SELECT * FROM user_profiles WHERE id = :profileId")
    suspend fun getProfileById(profileId: Long): UserProfile?
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertProfile(profile: UserProfile): Long
    
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertProfiles(profiles: List<UserProfile>)
    
    @Update
    suspend fun updateProfile(profile: UserProfile)
    
    @Delete
    suspend fun deleteProfile(profile: UserProfile)
    
    @Query("DELETE FROM user_profiles WHERE id = :profileId")
    suspend fun deleteProfileById(profileId: Long)
    
    @Query("UPDATE user_profiles SET is_active = 0")
    suspend fun deactivateAllProfiles()
    
    @Query("UPDATE user_profiles SET is_active = 1 WHERE id = :profileId")
    suspend fun activateProfile(profileId: Long)
    
    @Query("SELECT COUNT(*) FROM user_profiles")
    suspend fun getProfileCount(): Int
}
